rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java

```diff
@@ -575,15 +574,15 @@ public void setWork(MinerWork work) {
     }
 
     public MinerWork updateGetWork(@Nonnull final Block block, @Nonnull final boolean notify) {
-        Sha3Hash blockMergedMiningHash = new Sha3Hash(block.getHashForMergedMining());
+        Keccak256 blockMergedMiningHash = block.getHashForMergedMining();
 
         BigInteger targetBI = DifficultyUtils.difficultyToTarget(block.getDifficultyBI());
         byte[] targetUnknownLengthArray = targetBI.toByteArray();
         byte[] targetArray = new byte[32];
         System.arraycopy(targetUnknownLengthArray, 0, targetArray, 32 - targetUnknownLengthArray.length, targetUnknownLengthArray.length);
 
         logger.debug("Sending work for merged mining. Hash: {}", block.getShortHashForMergedMining());
-        return new MinerWork(TypeConverter.toJsonHex(blockMergedMiningHash.getBytes()), TypeConverter.toJsonHex(targetArray), String.valueOf(block.getFeesPaidToMiner()), notify, TypeConverter.toJsonHex(block.getParentHash()));
+        return new MinerWork(TypeConverter.toJsonHex(blockMergedMiningHash.getBytes()), TypeConverter.toJsonHex(targetArray), String.valueOf(block.getFeesPaidToMiner()), notify, TypeConverter.toJsonHex(block.getParentHash().getBytes()));
```

Comment:
What's the difference between the jsonHex and the string repr of the hash? We can add a "toJsonString" or something like that to avoid the getBytes() call